core/unix/loader.c

```diff
@@ -741,17 +741,55 @@ privload_search_rpath(privmod_t *mod, bool runpath, const char *name,
                     len = sep - list;
                 /* support $ORIGIN expansion to lib's current directory */
                 origin = strstr(list, RPATH_ORIGIN);
+#    ifdef CLIENT_INTERFACE
+                char path[MAXIMUM_PATH];
+#    endif
                 if (origin != NULL && origin < list + len) {
                     size_t pre_len = origin - list;
+#    ifdef CLIENT_INTERFACE
+                    snprintf(path, MAXIMUM_PATH, "%.*s%.*s%.*s", pre_len, list,
+                             moddir_len, mod->path,
+                             /* the '/' should already be here */
+                             len - strlen(RPATH_ORIGIN) - pre_len,
+                             origin + strlen(RPATH_ORIGIN));
```

Comment:
style: missing NULL_T*

Reply:
line 769?

Reply:
Since I did [this](https://github.com/DynamoRIO/dynamorio/pull/3854#discussion_r329177265), I had to move the NULL_T here as well.